src/WidgetBase.ts

```diff
@@ -429,26 +452,46 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		let nodes = Array.isArray(node) ? [ ...node ] : [ node ];
 		while (nodes.length) {
 			const node = nodes.pop();
-			if (isHNode(node)) {
-				if (node.properties.key) {
-					node.properties.afterCreate = this._afterCreateCallback;
-					node.properties.afterUpdate = this._afterUpdateCallback;
+			if (isHNode(node) || isWNode(node)) {
+				node.properties = node.properties || {};
+				if (isHNode(node)) {
+					if (node.properties.key) {
+						node.properties.afterCreate = this._afterCreateCallback;
+						node.properties.afterUpdate = this._afterUpdateCallback;
+					}
+					if (node.properties.bind === undefined) {
+						(<any> node.properties).bind = this;
+					}
 				}
+				else {
+					const { defaultRegistry, registry } = this.getCoreProperties(node.properties);
+					node.coreProperties = node.coreProperties || {};
+
+					if (node.coreProperties.bind === undefined) {
+						node.coreProperties.bind = this;
+					}
+					node.coreProperties.defaultRegistry = defaultRegistry;
```

Comment:
do you think we should make `coreProperties` a symbol? or bad idea?

Reply:
I'm not sure what it would give?

Reply:
i guess it would make it transparent to the end user. but i think we agreed that didn't matter anyway.